diff --git a/target/linux/brcm2708/base-files/etc/board.d/02_network b/target/linux/brcm2708/base-files/etc/board.d/02_network
index 7642c06..372302f 100755
--- a/target/linux/brcm2708/base-files/etc/board.d/02_network
+++ b/target/linux/brcm2708/base-files/etc/board.d/02_network
@@ -15,7 +15,14 @@ rpi-b |\
 rpi-b-plus |\
 rpi-2-b |\
 rpi-3-b)
-	ucidef_set_interface_lan "eth0"
+	ucidef_set_interface_lan "eth0" "dhcp"
+
+	ucidef_set_interface_raw "fallback" "eth0" "static"
+	json_select_object network
+	  json_select_object fallback
+	  json_add_string ipaddr "192.168.200.1"
+	  json_select_object ..
+	json_select_object ..
 	;;
 esac
 
diff --git a/target/linux/brcm2708/base-files/etc/diag.sh b/target/linux/brcm2708/base-files/etc/diag.sh
index 2a866e4..db902ab 100644
--- a/target/linux/brcm2708/base-files/etc/diag.sh
+++ b/target/linux/brcm2708/base-files/etc/diag.sh
@@ -8,12 +8,12 @@ set_state() {
 	case "$(brcm2708_board_name)" in
 	rpi-b |\
 	rpi-cm |\
-	rpi-zero)
+	rpi-zero |\
+	rpi-3-b)
 		status_led="led0"
 		;;
 	rpi-b-plus |\
-	rpi-2-b |\
-	rpi-3-b)
+	rpi-2-b)
 		status_led="led1"
 		;;
 	esac
@@ -29,6 +29,7 @@ set_state() {
 		status_led_blink_preinit_regular
 		;;
 	done)
+		status_led_off
 		status_led_on
 		;;
 	esac
diff --git a/target/linux/brcm2708/base-files/lib/upgrade/platform.sh b/target/linux/brcm2708/base-files/lib/upgrade/platform.sh
index 1d4b694..c293f0f 100644
--- a/target/linux/brcm2708/base-files/lib/upgrade/platform.sh
+++ b/target/linux/brcm2708/base-files/lib/upgrade/platform.sh
@@ -1,18 +1,50 @@
+USE_REFRESH=1
+RAMFS_COPY_BIN=/usr/sbin/partprobe
+
 platform_check_image() {
-	# i know no way to verify the image
-	return 0;
+	[ "$ARGC" -gt 1 ] && return 1
+
+	case "$(get_magic_word "$1")" in
+		0000) return 0;;
+		*)
+			echo "Invalid image type"
+			return 1
+		;;
+	esac
+}
+
+platform_refresh_partitions() {
+	partprobe /dev/mmcblk0
+}
+
+platform_copy_config() {
+	# remounting corrupts the filesystem, rely on config_restore
+	return 0
+	mount -t ext4 -o ro,noatime /dev/mmcblk0p2 /mnt && \
+	  mount -t ext4 -o remount,rw,noatime /dev/mmcblk0p2 /mnt && \
+	  cp -af "$CONF_TAR" /mnt/ && \
+	  umount /mnt
 }
 
 platform_do_upgrade() {
 	sync
-	get_image "$1" | dd of=/dev/mmcblk0 bs=2M conv=fsync
+	# If not saving the configuration, wipe the config partition
+	[ $SAVE_CONFIG -eq 0 -a -b /dev/mmcblk0p4 ] && {
+		# Should do mkfs.ext4 here but isn't on ramdisk
+		  dd if=/dev/zero of=/dev/mmcblk0p4 bs=1M count=1 conv=fsync
+	}
+	dd if=/dev/zero of=/dev/mmcblk0 bs=4M count=10 conv=fsync
+	get_image "$@" | dd of=/dev/mmcblk0 bs=4M conv=fsync
 	sleep 1
 }
 
-platform_copy_config() {
-	mkdir -p /boot
-	[ -f /boot/kernel.img ] || mount -t vfat -o rw,noatime /dev/mmcblk0p1 /boot
-	cp -af "$CONF_TAR" /boot/
-	sync
-	umount /boot
+bcm2708_umount_all() {
+	cd /tmp
+	echo powersave > /sys/devices/system/cpu/cpu0/cpufreq/scaling_governor
+	lmclient LMDC,0
+	umount /dev/mmcblk0p1
+	umount /dev/mmcblk0p4
+	umount /sys/kernel/debug
 }
+
+append sysupgrade_pre_upgrade bcm2708_umount_all
diff --git a/target/linux/brcm2708/image/Makefile b/target/linux/brcm2708/image/Makefile
old mode 100644
new mode 100755
index e7089c6..a86fb88
--- a/target/linux/brcm2708/image/Makefile
+++ b/target/linux/brcm2708/image/Makefile
@@ -30,6 +30,7 @@ define Build/boot-img
 	mcopy -i $@.boot cmdline.txt ::
 	mcopy -i $@.boot config.txt ::
 	mcopy -i $@.boot $(word 1,$^) ::kernel.img
+	#mcopy -i $@.boot /home/bmayland/trunk-openwrt/build_dir/target-arm_cortex-a53+neon-vfpv4_musl-1.1.14_eabi/linux-brcm2708_bcm2710/kernel7-bry.img ::kernel7.img
 	$(foreach dts,$(shell echo $(DEVICE_DTS)),mcopy -i $@.boot $(DTS_DIR)/$(dts).dtb ::;)
 	mmd -i $@.boot ::/overlays
 	mcopy -i $@.boot $(DTS_DIR)/overlays/*.dtbo ::/overlays/
@@ -74,13 +75,13 @@ endif
 ### BCM2709/BCM2836 ###
 ifeq ($(SUBTARGET),bcm2709)
   # Raspberry Pi 2 Model B
-  $(eval $(call bcm27xx,RaspberryPi_2,rpi-2,bcm2709-rpi-2-b))
+  $(eval $(call bcm27xx,RaspberryPi_2,rpi-2,bcm2709-rpi-2-b bcm2710-rpi-3-b))
 endif
 
 ### BCM2710/BCM2837 ###
 ifeq ($(SUBTARGET),bcm2710)
   # Raspberry Pi 3 Model B
-  $(eval $(call bcm27xx,RaspberryPi_3,rpi-3,bcm2710-rpi-3-b))
+  $(eval $(call bcm27xx,RaspberryPi_3,rpi-3,bcm2709-rpi-2-b bcm2710-rpi-3-b))
 endif
 
 $(eval $(call BuildImage))
diff --git a/target/linux/brcm2708/image/cmdline.txt b/target/linux/brcm2708/image/cmdline.txt
index 24f4f0c..766bae3 100644
--- a/target/linux/brcm2708/image/cmdline.txt
+++ b/target/linux/brcm2708/image/cmdline.txt
@@ -1 +1 @@
-dwc_otg.lpm_enable=0 console=serial0,115200 kgdboc=serial0,115200 console=tty1 root=/dev/mmcblk0p2 rootfstype=ext4 rootwait
+dwc_otg.lpm_enable=0 console=tty1 root=/dev/mmcblk0p2 rootfstype=ext4 rootwait
diff --git a/target/linux/brcm2708/image/config.txt b/target/linux/brcm2708/image/config.txt
index e80d980..84e4092 100644
--- a/target/linux/brcm2708/image/config.txt
+++ b/target/linux/brcm2708/image/config.txt
@@ -949,3 +949,25 @@ dtparam=random=on,watchdog=on,audio=on,i2c0=on,i2c1=on,spi=on
 #dtoverlay=rpi-dac
 #dtoverlay=rpi-proto
 #dtoverlay=rra-digidac1-wm8741-audio
+
+# RPi3 defaults UART off because this requires core_freq=250 so baud clock works correctly
+enable_uart=1
+core_freq=250
+
+##
+## wifi configuration
+##
+# SSID (network name)
+#wifi_ssid=heatermeter
+# Password for encryption
+#wifi_password=password
+# Encryption mode psk2 (WPA2-PSK default), psk, wep, none
+#wifi_encryption=psk2
+# Mode ap (Access Point) or sta (Client mode, default). Must be lowercase!
+#wifi_mode=ap
+# wifi channel, only used in AP mode
+#wifi_channel=6
+
+hdmi_cvt=1024 600 60 5
+hdmi_blanking=1
+
