--- /tmp/base-files-network/files/lib/network/config.sh	2011-10-22 13:43:38.138311346 -0400
+++ base-files-network/files/lib/network/config.sh	2011-10-22 13:53:51.322778137 -0400
@@ -378,6 +378,8 @@
 
 			[ -z "$ipaddr" ] || \
 				$DEBUG ifconfig "$iface" "$ipaddr" ${netmask:+netmask "$netmask"}
+			[ -n "$hostname" ] || hostname=$(uci -q get system.@system[0].hostname)
+			[ "$hostname" = "none" ] && hostname=""
 
 			# additional request options
 			local opt dhcpopts
