Index: base-files-network/files/lib/network/config.sh
===================================================================
--- base-files-network/files/lib/network/config.sh	(revision 29665)
+++ base-files-network/files/lib/network/config.sh	(working copy)
@@ -384,6 +384,8 @@
 
 			[ -z "$ipaddr" ] || \
 				$DEBUG ifconfig "$iface" "$ipaddr" ${netmask:+netmask "$netmask"}
+			[ -n "$hostname" ] || hostname=$(uci -q get system.@system[0].hostname)
+			[ "$hostname" = "none" ] && hostname=""
 
 			# additional request options
 			local opt dhcpopts daemonize
