Index: package/dnsmasq/files/dnsmasq.init
===================================================================
--- package/dnsmasq/files/dnsmasq.init	(revision 36720)
+++ package/dnsmasq/files/dnsmasq.init	(working copy)
@@ -464,8 +464,8 @@
 	include /lib/network
 	scan_interfaces
 
-	local lanaddr
-	config_get lanaddr "lan" ipaddr
+	local lanifname
+	config_get lanifname "lan" ifname
 
 	config_load dhcp
 
@@ -502,9 +502,13 @@
 	echo >> $CONFIGFILE
 
 	# add own hostname
-	[ $ADD_LOCAL_HOSTNAME -eq 1 ] && [ -n "$lanaddr" ] && {
+	[ $ADD_LOCAL_HOSTNAME -eq 1 ] && [ -n "$lanifname" ] && {
 		local hostname="$(uci_get system.@system[0].hostname)"
 		dhcp_domain_add "" "${hostname:-OpenWrt}" "$lanaddr"
+		[ "${hostname%.*}" == "$hostname" ] && {
+			hostname="${hostname:-OpenWrt}.${DOMAIN:-lan}"
+		}
+		xappend "--interface-name=$hostname, $lanifname"
 	}
 
 	service_start /usr/sbin/dnsmasq -C $CONFIGFILE && {
