--- /tmp/luci-trunk+svn8147/libs/web/luasrc/dispatcher.lua	2011-11-30 07:50:32.000000000 -0500
+++ luci-trunk+svn8147/libs/web/luasrc/dispatcher.lua	2012-01-07 10:52:12.000000000 -0500
@@ -677,12 +677,13 @@
 
 	if not c then
 		local last = table.remove(path)
+		local inreq = context.path[#path+1] == last
 		local parent = _create_node(path)
 
 		c = {nodes={}, auto=true}
 		-- the node is "in request" if the request path matches
 		-- at least up to the length of the node path
-		if parent.inreq and context.path[#path+1] == last then
+		if inreq and parent.inreq then
 		  c.inreq = true
 		end
 		parent.nodes[last] = c
