diff -u /tmp/luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/network.lua luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/network.lua
--- /tmp/luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/network.lua	2011-10-18 09:20:43.000000000 -0400
+++ luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/network.lua	2011-10-25 09:35:40.120691487 -0400
@@ -16,6 +16,15 @@
 module("luci.controller.admin.network", package.seeall)
 
 function index()
+	local page
+
+	page = node("admin", "network")
+	page.target = alias("admin", "network", "network")
+	page.title = _("Network")
+	page.order = 50
+	page.index  = true
+	if not page.inreq then return end
+
 	local uci = require("luci.model.uci").cursor()
 	local net = require "luci.model.network".init(uci)
 	local has_wifi = nixio.fs.stat("/etc/config/wireless")
@@ -28,14 +37,6 @@
 		end
 	)
 
-	local page
-
-	page = node("admin", "network")
-	page.target = alias("admin", "network", "network")
-	page.title  = _("Network")
-	page.order  = 50
-	page.index  = true
-
 	if has_switch then
 		page  = node("admin", "network", "vlan")
 		page.target = cbi("admin_network/vlan")
diff -u /tmp/luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/services.lua luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/services.lua
--- /tmp/luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/services.lua	2011-08-12 09:16:27.000000000 -0400
+++ luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/services.lua	2011-10-25 09:33:07.382269466 -0400
@@ -16,15 +16,15 @@
 
 function index()
 	local page
-
-	page        = node("admin", "services", "crontab")
-	page.target = form("admin_services/crontab")
-	page.title  = _("Scheduled Tasks")
-	page.order  = 50
-
 	page        = node("admin", "services")
 	page.target = template("admin_services/index")
 	page.title  = _("Services")
 	page.order  = 40
 	page.index  = true
+	if not page.inreq then return end
+
+	page        = node("admin", "services", "crontab")
+	page.target = form("admin_services/crontab")
+	page.title  = _("Scheduled Tasks")
+	page.order  = 50
 end
diff -u /tmp/luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/status.lua luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/status.lua
--- /tmp/luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/status.lua	2011-10-15 01:10:58.000000000 -0400
+++ luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/status.lua	2011-10-24 16:44:07.401632223 -0400
@@ -16,7 +16,11 @@
 module("luci.controller.admin.status", package.seeall)
 
 function index()
-	entry({"admin", "status"}, alias("admin", "status", "overview"), _("Status"), 20).index = true
+	local tnode
+	tnode = entry({"admin", "status"}, alias("admin", "status", "overview"), _("Status"), 20)
+	tnode.index = true
+	if not tnode.inreq then return end
+
 	entry({"admin", "status", "overview"}, template("admin_status/index"), _("Overview"), 1)
 	entry({"admin", "status", "iptables"}, call("action_iptables"), _("Firewall"), 2).leaf = true
 	entry({"admin", "status", "routes"}, template("admin_status/routes"), _("Routes"), 3)
diff -u /tmp/luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/system.lua luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/system.lua
--- /tmp/luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/system.lua	2011-10-15 01:27:06.000000000 -0400
+++ luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/system.lua	2011-10-24 16:44:31.214017303 -0400
@@ -16,7 +16,11 @@
 module("luci.controller.admin.system", package.seeall)
 
 function index()
-	entry({"admin", "system"}, alias("admin", "system", "system"), _("System"), 30).index = true
+	local tnode
+	tnode = entry({"admin", "system"}, alias("admin", "system", "system"), _("System"), 30)
+	tnode.index = true
+	if not tnode.inreq then return end
+
 	entry({"admin", "system", "system"}, cbi("admin_system/system"), _("System"), 1)
 	entry({"admin", "system", "admin"}, cbi("admin_system/admin"), _("Administration"), 2)
 
diff -u /tmp/luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/uci.lua luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/uci.lua
--- /tmp/luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/uci.lua	2011-08-12 09:16:27.000000000 -0400
+++ luci-trunk+svn7768/modules/admin-full/luasrc/controller/admin/uci.lua	2011-10-24 16:42:11.427756442 -0400
@@ -16,10 +16,13 @@
 module("luci.controller.admin.uci", package.seeall)
 
 function index()
+	local tnode
+	tnode = entry({"admin", "uci"}, nil, _("Configuration"))
+	if not tnode.inreq then return end
+
 	local redir = luci.http.formvalue("redir", true) or
 	  luci.dispatcher.build_url(unpack(luci.dispatcher.context.request))
 
-	entry({"admin", "uci"}, nil, _("Configuration"))
 	entry({"admin", "uci", "changes"}, call("action_changes"), _("Changes"), 40).query = {redir=redir}
 	entry({"admin", "uci", "revert"}, call("action_revert"), _("Revert"), 30).query = {redir=redir}
 	entry({"admin", "uci", "apply"}, call("action_apply"), _("Apply"), 20).query = {redir=redir}
