diff -u /tmp/luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/index.lua luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/index.lua
--- /tmp/luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/index.lua	2011-10-25 23:04:18.000000000 -0400
+++ luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/index.lua	2012-01-07 11:13:25.000000000 -0500
@@ -29,6 +29,7 @@
 	page.sysauth_authenticator = "htmlauth"
 	page.ucidata = true
 	page.index = true
+	if not page.inreq then return end
 
 	-- Empty services menu to be populated by addons
 	entry({"admin", "services"}, firstchild(), _("Services"), 40).index = true
diff -u /tmp/luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/system.lua luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/system.lua
--- /tmp/luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/system.lua	2011-12-20 12:35:50.000000000 -0500
+++ luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/system.lua	2012-01-07 11:14:57.000000000 -0500
@@ -16,7 +16,10 @@
 module("luci.controller.admin.system", package.seeall)
 
 function index()
-	entry({"admin", "system"}, alias("admin", "system", "system"), _("System"), 30).index = true
+	local page = entry({"admin", "system"}, alias("admin", "system", "system"), _("System"), 30)
+	page.index = true
+	if not page.inreq then return end
+
 	entry({"admin", "system", "system"}, cbi("admin_system/system"), _("System"), 1)
 	entry({"admin", "system", "clock_status"}, call("action_clock_status"))
 
diff -u /tmp/luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/uci.lua luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/uci.lua
--- /tmp/luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/uci.lua	2011-08-12 09:16:27.000000000 -0400
+++ luci-trunk+svn8147/modules/admin-full/luasrc/controller/admin/uci.lua	2012-01-07 11:11:08.000000000 -0500
@@ -16,10 +16,11 @@
 module("luci.controller.admin.uci", package.seeall)
 
 function index()
+	local page = entry({"admin", "uci"}, nil, _("Configuration"))
+	if not page.inreq then return end
+
 	local redir = luci.http.formvalue("redir", true) or
 	  luci.dispatcher.build_url(unpack(luci.dispatcher.context.request))
-
-	entry({"admin", "uci"}, nil, _("Configuration"))
 	entry({"admin", "uci", "changes"}, call("action_changes"), _("Changes"), 40).query = {redir=redir}
 	entry({"admin", "uci", "revert"}, call("action_revert"), _("Revert"), 30).query = {redir=redir}
 	entry({"admin", "uci", "apply"}, call("action_apply"), _("Apply"), 20).query = {redir=redir}
