--- luci-trunk+svn9411/libs/web/luasrc/dispatcher.lua.orig	2012-11-16 10:26:50.564328881 -0500
+++ luci-trunk+svn9411/libs/web/luasrc/dispatcher.lua	2012-11-16 10:29:02.430266381 -0500
@@ -366,21 +366,21 @@
 
 		if not util.contains(accs, user) then
 			if authen then
-				ctx.urltoken.stok = nil
+				if ctx.urltoken.stok == nil then
+					ctx.urltoken.stok = luci.sys.uniqueid(16)
+				end
 				local user, sess = authen(luci.sys.user.checkpasswd, accs, def)
 				if not user or not util.contains(accs, user) then
 					return
 				else
 					local sid = sess or luci.sys.uniqueid(16)
 					if not sess then
-						local token = luci.sys.uniqueid(16)
 						sauth.reap()
 						sauth.write(sid, {
 							user=user,
-							token=token,
+							token=ctx.urltoken.stok,
 							secret=luci.sys.uniqueid(16)
 						})
-						ctx.urltoken.stok = token
 					end
 					luci.http.header("Set-Cookie", "sysauth=" .. sid.."; path="..build_url())
 					ctx.authsession = sid
--- luci-trunk+svn9411/modules/admin-core/luasrc/view/sysauth.htm.orig	2012-11-16 10:26:08.787707601 -0500
+++ luci-trunk+svn9411/modules/admin-core/luasrc/view/sysauth.htm	2012-11-16 10:26:23.883932540 -0500
@@ -15,7 +15,7 @@
 
 <%+header%>
 
-<form method="post" action="<%=pcdata(luci.http.getenv("REQUEST_URI"))%>">
+<form method="post" action="<%=REQUEST_URI%>">
 	<div class="cbi-map">
 		<h2><a id="content" name="content"><%:Authorization Required%></a></h2>
 		<div class="cbi-map-descr">
