diff --git a/modules/luci-base/luasrc/dispatcher.lua b/modules/luci-base/luasrc/dispatcher.lua
index e4f77f1..d5f75bf 100644
--- a/modules/luci-base/luasrc/dispatcher.lua
+++ b/modules/luci-base/luasrc/dispatcher.lua
@@ -386,8 +386,8 @@ function dispatch(request)
 			local pass = http.getenv("HTTP_AUTH_PASS")
 
 			if user == nil and pass == nil then
-				user = http.formvalue("luci_username")
-				pass = http.formvalue("luci_password")
+				user = http.formvalue("luci_username") or http.formvalue("username")
+				pass = http.formvalue("luci_password") or http.formvalue("password")
 			end
 
 			sid, sdat = session_setup(user, pass, allowed_users)
