--- package/iwinfo.orig/src/iwinfo_nl80211.c	2012-11-14 14:51:59.279459532 -0500
+++ package/iwinfo/src/iwinfo_nl80211.c	2012-11-14 14:59:18.129642576 -0500
@@ -1766,7 +1766,13 @@
 				goto out;
 
 			if (!iwinfo_ifmac(res))
+			{
+				/* 8192cu: If virtual interfaces are not supported
+				 * run the scan on the main interface */
+				iwinfo_ifup(ifname);
+				nl80211_get_scanlist_nl(ifname, buf, len);
 				goto out;
+			}
 
 			/* if we can take the new interface up, the driver supports an
 			 * additional interface and there's no need to tear down the ap */
