Index: package/iwinfo/src/iwinfo_nl80211.c
===================================================================
--- package/iwinfo/src/iwinfo_nl80211.c	(revision 36655)
+++ package/iwinfo/src/iwinfo_nl80211.c	(working copy)
@@ -1940,7 +1940,13 @@
 				goto out;
 
 			if (!iwinfo_ifmac(res))
+			{
+				/* 8192cu: If virtual interfaces are not supported
+				 * run the scan on the main interface */
+				iwinfo_ifup(ifname);
+				nl80211_get_scanlist_nl(ifname, buf, len);
 				goto out;
+			}
 
 			/* if we can take the new interface up, the driver supports an
 			 * additional interface and there's no need to tear down the ap */
