--- package/iwinfo/src/iwinfo_nl80211.orig.c	2013-05-26 08:47:36.471094286 -0400
+++ package/iwinfo/src/iwinfo_nl80211.c	2013-05-26 08:47:42.611094343 -0400
@@ -1766,7 +1766,13 @@ int nl80211_get_scanlist(const char *ifn
 				goto out;
 
 			if (!iwinfo_ifmac(res))
+			{
+				/* 8192cu: If virtual interfaces are not supported
+				 * run the scan on the main interface */
+				iwinfo_ifup(ifname);
+				nl80211_get_scanlist_nl(ifname, buf, len);
 				goto out;
+			}
 
 			/* if we can take the new interface up, the driver supports an
 			 * additional interface and there's no need to tear down the ap */
