--- luci-0.11.1/libs/lucid-http/luasrc/lucid/http/server.lua.orig	2014-06-27 09:25:42.702712331 -0400
+++ luci-0.11.1/libs/lucid-http/luasrc/lucid/http/server.lua	2014-06-27 09:26:29.954710852 -0400
@@ -195,13 +195,13 @@
 	-- SCRIPT_NAME
 	request.env.SCRIPT_NAME = ""
 
-	-- Call URI part
-	request.env.PATH_INFO = uri
-	
 	if self.default and uri == "/" then
-		return 302, {Location = self.default}
+		uri = self.default
 	end
 
+	-- Call URI part
+	request.env.PATH_INFO = uri
+
 	for k, h in pairs(self.handlers) do
 		if #k > hlen then
 			if uri == k or (uri:sub(1, #k) == k and uri:byte(#k+1) == sc) then
