--- luci-0.11.1/libs/lucid-http/luasrc/lucid/http/server.lua.orig	2014-06-27 11:19:28.114498630 -0400
+++ luci-0.11.1/libs/lucid-http/luasrc/lucid/http/server.lua	2014-06-27 11:19:57.290497717 -0400
@@ -461,7 +461,7 @@
 		end
 
 		-- Uncomment this to disable keep-alive
-		close = close or env.config.nokeepalive
+		close = close or env.config.nokeepalive == "1"
 	
 		if message.env.REQUEST_METHOD == "GET"
 		or message.env.REQUEST_METHOD == "HEAD" then
