diff --git a/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi_add.lua b/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi_add.lua
index 96b8b4d..64ce572 100644
--- a/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi_add.lua
+++ b/modules/luci-mod-admin-full/luasrc/model/cbi/admin_network/wifi_add.lua
@@ -157,6 +157,11 @@ function newnet.parse(self, section)
 			uci:save("network")
 			uci:save("firewall")
 
+			if net:proto() == "dhcp" then
+				uci:set("dhcp", net:name() .. ".ignore", "1")
+				uci:save("dhcp")
+			end
+
 			luci.http.redirect(wnet:adminlink())
 		end
 	end
