diff --git a/package/network/config/netifd/files/lib/netifd/proto/dhcp.sh b/package/network/config/netifd/files/lib/netifd/proto/dhcp.sh
index ea02d68..634d33d 100755
--- a/package/network/config/netifd/files/lib/netifd/proto/dhcp.sh
+++ b/package/network/config/netifd/files/lib/netifd/proto/dhcp.sh
@@ -52,6 +52,7 @@ proto_dhcp_setup() {
 	[ "$delegate" = "0" ] && proto_export "IFACE6RD_DELEGATE=0"
 	# Request classless route option (see RFC 3442) by default
 	[ "$classlessroute" = "0" ] || append dhcpopts "-O 121"
+	[ -n "$hostname" ] || hostname=$(uci_get system @system[0] hostname)
 
 	proto_export "INTERFACE=$config"
 	proto_run_command "$config" udhcpc \
