--- feeds/packages/libs/clearsilver/Makefile.orig	2016-01-03 12:40:15.767363426 -0500
+++ feeds/packages/libs/clearsilver/Makefile	2016-01-03 12:41:07.647263303 -0500
@@ -31,7 +31,7 @@
   CATEGORY:=Libraries
   TITLE:=ClearSilver template system
   URL:=http://www.clearsilver.net/
-  MAINTAINER:=Raphaël HUCK <rhk@cksum.org>
+  MAINTAINER:=Raphael HUCK <rhk@cksum.org>
   DEPENDS:=+CLEARSILVER_ENABLE_COMPRESSION:zlib +CLEARSILVER_ENABLE_GETTEXT:libintl
 endef
 
