--- /tmp/avrdude-5.8/avrdude.conf.in	2009-07-10 18:22:22.000000000 -0400
+++ avrdude-5.8/avrdude.conf.in	2011-05-14 08:58:52.150631398 -0400
@@ -303,18 +303,22 @@
 # Overall avrdude defaults
 #
 default_parallel   = "@DEFAULT_PAR_PORT@";
+@LINKMETER_SAVE_BEGIN@
 default_serial     = "@DEFAULT_SER_PORT@";
+@LINKMETER_SAVE_END@
 
 
 #
 # PROGRAMMER DEFINITIONS
 #
 
+@LINKMETER_SAVE_BEGIN@
 programmer
   id    = "arduino";
   desc  = "Arduino";
   type  = arduino;
 ;
+@LINKMETER_SAVE_END@
 
 programmer
   id    = "avrisp";
@@ -8460,7 +8464,7 @@
 #------------------------------------------------------------
 # ATmega328P
 #------------------------------------------------------------
-
+@LINKMETER_SAVE_BEGIN@
 part
     id			= "m328p";
     desc		= "ATMEGA328P";
@@ -8644,6 +8648,7 @@
 	       "x x x x x x a1 a0 o o o o o o o o";
     ;
 ;
+@LINKMETER_SAVE_END@
 
 #------------------------------------------------------------
 # ATtiny2313
--- /tmp/avrdude-5.8/configure.ac	2011-05-14 09:03:35.124357721 -0400
+++ avrdude-5.8/configure.ac	2011-05-14 09:03:49.074541254 -0400
@@ -239,6 +239,7 @@
 else
 	confsubst="-e /^@HAVE_PARPORT_BEGIN@/,/^@HAVE_PARPORT_END@/d"
 fi
+confsubst="$confsubst -e /^@LINKMETER_SAVE_BEGIN@/,/^@LINKMETER_SAVE_END@/!d -e /^@LINKMETER_/d"
 AC_SUBST(confsubst)
 
 # See if we need to drop into the windows subdir.
