--- /tmp/rrdtool-1.4.5/bindings/lua/rrdlua.c	2009-05-26 11:24:26.000000000 -0400
+++ rrdtool-1.4.5/bindings/lua/rrdlua.c	2011-05-21 14:57:01.836173403 -0400
@@ -66,7 +66,7 @@
   for (i=1; i<argc; i++) {
     /* accepts string or number */
     if (lua_isstring(L, i) || lua_isnumber(L, i)) {
-      if (!(argv[i] = strdup(lua_tostring (L, i)))) {
+      if (!(argv[i] = lua_tostring (L, i))) {
         /* raise an error and never return */
         luaL_error(L, "%s - error duplicating string area for arg #%d",
                    cmd, i);
